test/pkg/ptptesthelper/ptptesthelper.go

```diff
@@ -272,53 +272,107 @@ func CheckSlaveSyncWithMaster(fullConfig testconfig.TestConfig) {
 	dummyLabels := map[string]string{}
-	rebootDaemonSetRunningPods, err := k8sPriviledgedDs.CreateDaemonSet(pkg.RebootDaemonSetName, pkg.RebootDaemonSetNamespace, pkg.RebootDaemonSetContainerName, imageWithVersion, dummyLabels, pkg.TimeoutIn5Minutes)
+	daemonSetRunningPods, err := k8sPriviledgedDs.CreateDaemonSet(daemonsetName, daemonsetNamespace, daemonsetContainerName, imageWithVersion, dummyLabels, pkg.TimeoutIn5Minutes)
+
 	if err != nil {
 		logrus.Errorf("error : +%v\n", err.Error())
 	}
-	nodeToPodMapping := make(map[string]corev1.Pod)
-	for _, dsPod := range rebootDaemonSetRunningPods.Items {
-		nodeToPodMapping[dsPod.Spec.NodeName] = dsPod
-	}
+	return daemonSetRunningPods
+}
 
-	// 2. Get the slave config, restart the slave node
-	slavePtpConfig := fullConfig.DiscoveredClockUnderTestPtpConfig
-	restartedNodes := make([]string, len(rebootDaemonSetRunningPods.Items))
+func RecoverySlaveNetworkOutage(fullConfig testconfig.TestConfig) {
+	logrus.Info("Recovery network outage begins ...........")
+
+	// Get a slave pod
+	slavePod, err := ptphelper.GetPTPPodWithPTPConfig((*ptpv1.PtpConfig)(fullConfig.DiscoveredClockUnderTestPtpConfig))
+	if err != nil {
+		logrus.Error("Could not determine ptp daemon pod selected by ptpconfig")
+	}
+	slavePodNodeName := slavePod.Spec.NodeName
+	logrus.Info("slave node name is ", slavePodNodeName)
 
-	for _, recommend := range slavePtpConfig.Spec.Recommend {
-		matches := recommend.Match
+	// Get the pod from ptp test daemonset set on the slave node
+	networkRecoveryDaemonSetRunningPods := CreatePtpTestPrivilegedDaemonSet(pkg.RecoveryNetworkOutageDaemonSetName, pkg.RecoveryNetworkOutageDaemonSetNamespace, pkg.RecoveryNetworkOutageDaemonSetContainerName)
+	var networkOutageRecoveryDaemonsetPod corev1.Pod
 
-		for _, match := range matches {
-			nodeLabel := *match.NodeLabel
-			// Get all nodes with this node label
-			nodes, err := client.Client.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{LabelSelector: nodeLabel})
-			Expect(err).NotTo(HaveOccurred())
+	Expect(len(networkRecoveryDaemonSetRunningPods.Items)).To(BeNumerically(">", 0), "no damonset pods found in the namespace "+pkg.RecoveryNetworkOutageDaemonSetNamespace)
 
-			// Restart the node
-			for _, node := range nodes.Items {
-				nodeshelper.RebootNode(nodeToPodMapping[node.Name], node)
-				restartedNodes = append(restartedNodes, node.Name)
-			}
+	for _, dsPod := range networkRecoveryDaemonSetRunningPods.Items {
+		if dsPod.Spec.NodeName == slavePodNodeName {
+			networkOutageRecoveryDaemonsetPod = dsPod
+			break
 		}
 	}
-	logrus.Printf("Restarted nodes %v", restartedNodes)
+	logrus.Info("outage recovery pod name is ", networkOutageRecoveryDaemonsetPod.Name)
+
+	// Get the list of network interfaces on the slave node
+	slaveIf := ptpv1.GetInterfaces((ptpv1.PtpConfig)(*fullConfig.DiscoveredClockUnderTestPtpConfig), ptpv1.Slave)
+
+	// Toggle the interfaces
+	for _, ptpNodeInterface := range slaveIf {
```

Comment:
What is this testing?

Reply:
I mark this as resolved as we discussed over the calls.